subprojects/core/src/main/java/org/gradle/initialization/ParallelismBuildOptionFactory.java

```diff
@@ -56,34 +56,23 @@ public void applyTo(boolean value, ParallelismConfiguration settings, Origin ori
     public static class MaxWorkersOption extends StringBuildOption<ParallelismConfiguration> {
         public static final String GRADLE_PROPERTY = "org.gradle.workers.max";
         public static final String LONG_OPTION = "max-workers";
+        public static final String HINT = "must be a positive, non-zero, integer";
 
         public MaxWorkersOption() {
-            super(GRADLE_PROPERTY, CommandLineOptionConfiguration.create("max-workers", "Configure the number of concurrent workers Gradle is allowed to use.").incubating());
+            super(GRADLE_PROPERTY, CommandLineOptionConfiguration.create(LONG_OPTION, "Configure the number of concurrent workers Gradle is allowed to use.").incubating());
         }
 
         @Override
         public void applyTo(String value, ParallelismConfiguration settings, Origin origin) {
             try {
                 int workerCount = Integer.parseInt(value);
                 if (workerCount < 1) {
-                    handleInvalidValue(origin, value);
+                    origin.handleInvalidValue(GRADLE_PROPERTY, LONG_OPTION, value, HINT);
                 }
                 settings.setMaxWorkerCount(workerCount);
             } catch (NumberFormatException e) {
-                handleInvalidValue(origin, value);
+                origin.handleInvalidValue(GRADLE_PROPERTY, LONG_OPTION, value, HINT);
```

Comment:
`GRADLE_PROPERTY` is irrelevant for the command line option error message.

Reply:
Yes, I was trying to reduce `if`s. Now I've fixed it.